airflow/triggers/base.py

```diff
@@ -119,3 +119,34 @@ def __eq__(self, other):
         if isinstance(other, TriggerEvent):
             return other.payload == self.payload
         return False
+
+
+class BaseTaskEndEvent(TriggerEvent):
+    """Base event class to end the task without resuming on worker."""
+
+    task_instance_state: TaskInstanceState
+
+    def __init__(self, *, xcom_return: Any | None = None, other_xcom: dict[str, Any] | None = None, **kwargs):
+        if "payload" in kwargs:
+            raise ValueError("Param 'payload' not supported for this class.")
```

Comment:
Why is this needed? I imagine this should work:

```suggestion
    def __init__(self, *, xcom_return: Any | None = None, other_xcom: dict[str, Any] | None = None) -> None:
```

Reply:
Because the payload is determined by the state / controlled by the class.  User should not submit a payload.

Reply:
if we do not check, user might pass `payload` and it would be silently ignored

Reply:
I don’t think so? The suggested change removed `payload` altogether and should raise TypeError when it is provided. Or is the issue you want to allow `**kwargs`? But I don’t see any subclass using that either.

Reply:
Ah i see what you're saying...  Well it's in the base class.  Would you not say it's probably a good idea to include kwargs in the subclasses for future compat?